numpy/linalg/tests/test_linalg.py

```diff
@@ -845,6 +846,111 @@ def test_empty(self):
         assert_equal(norm(array([], dtype=self.dt)), 0.0)
         assert_equal(norm(atleast_2d(array([], dtype=self.dt))), 0.0)
 
+    def test_vector_types(self):
+        a = np.array([1, 0, 1])
+
+        exact_types = [
```

Comment:
I did initially using two separate for loops, but thought it was cleaner to fold them together and loose that precision in checking ints as it didn't seem to add much compared to the added complexity.